### DISCUSSION TITLE
no responses when read a line

### DISCUSSION THREAD

**Discussion by FatDang** [sshnet/SSH.NET#1500] [Q&A]
when I change ssh.net's version from 2023.0.1 to 2024.1.0 and run the code. It will be no responses  when I use stream.read to read a line      continuously . It seems that if the reader meet the empty content the code will be no responses. It's normal when I use sshnet version 2023.0.1. How can I do to fix this problem.Thanks!

**Comment by Rob-Hague** [comment]
Can you please show some example code?

**Reply by FatDang** [reply]
StreamReader reader = null;
reader = new StreamReader(sshCommand.OutputStream);

        public string ReadByEndOf()
        {
            string tmp = null;
            string str;
            for (int i = 0; i < 3; i++)
            {
                do
                {
                    str = reader.ReadLine();
                    if (!string.IsNullOrWhiteSpace(str))
                    {
                        tmp += str.Trim() + "\r\n";
                        //Console.WriteLine(str.Trim());
                    }
                } while (!reader.EndOfStream);
                Thread.Sleep(500);
            }


            return tmp;
        }

As showed above， I initial a reader and use the function "ReadByEndOf()" . When it goes to the step "str = reader.ReadLine();"  it will get no responses after several while loops.  I tried to trace the invocation and  found the problem occur in "ReadBuffer()"

**Comment by katomark** [comment]
I have same issue. I was update from version 2020 to 2025 and it has same issue. StreamShell.DataAvailable always true and never false, so it stuck  at reader.read(). Then I tried to use 2023 version  and my code working again. Is 2025 version has different behaviour? I'm in windows 11

**Reply by Rob-Hague** [reply]
Yes, ShellStream was rewritten in 2024.0.0. It fixed incorrect behaviour where Read would return 0 (EOF) before the channel was closed. See also #1657 and #1485

**Reply by katomark** [reply]
Thank you for your update @Rob-Hague. So now, I should use timeout when read data, CMIW.

**Reply by katomark** [reply]
just update to 2025, and use shellstream.read() instead. Now it's working well
